src/main/scala/cromwell/binding/values/WdlArray.scala

```diff
@@ -34,4 +37,10 @@ case class WdlArray(wdlType: WdlArrayType, value: Seq[WdlValue]) extends WdlValu
   override def collectAsSeq[T <: WdlValue](filterFn: PartialFunction[WdlValue, T]): Seq[T] = {
     value flatMap { _.collectAsSeq(filterFn) }
   }
-}
\ No newline at end of file
+
+  override def getHash(implicit hasher: FileHasher) = {
+    val hashedArray = value map { _.getHash }
+    val accumulatedHash = hashedArray.foldLeft("") { (acc, v) => acc + v }
+    (getClass.getCanonicalName+accumulatedHash).md5Sum
+  }
```

Comment:
whoops this is String concatenation not numerical addition, never mind... :frowning: